nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2182,17 +2184,26 @@ public Lock setLock(DocumentRef docRef) throws LockException {
         Document doc = resolveReference(docRef);
         // TODO: add a new permission named LOCK and use it instead of
         // WRITE_PROPERTIES
-        checkPermission(doc, WRITE_PROPERTIES);
+        try {
+            LockSecurityPolicy.setIgnorePolicy(true);
```

Comment:
Add a comment before this line explaining why we do this: `ignore the lock policy that would always deny a write on a locked document, because we want to have a specific LockException when the document is already locked by another user`